javav2/usecases/creating_fsa_app/src/main/java/com/example/fsa/services/S3Service.java

```diff
@@ -0,0 +1,63 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.fsa.services;
+
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class S3Service {
+
+    private static S3Client s3Client;
+
+    private static synchronized S3Client getS3Client() {
+        if (s3Client == null) {
+            s3Client = S3Client.builder()
+                .region(Region.US_EAST_1)
+                .build();
+        }
+        return s3Client;
+    }
+
+    // Put the audio file into the Amazon S3 bucket.
+    public String putAudio(InputStream is, String bucket, String key) throws IOException {
+        try {
+            byte[] bytes = inputStreamToBytes(is);
+            long contentLength = bytes.length;
+
+            // Create a PutObjectRequest with content length.
+            PutObjectRequest putObjectRequest = PutObjectRequest.builder()
+                .bucket(bucket)
+                .contentType("audio/mp3")
+                .contentLength(contentLength)
+                .key(key)
+                .build();
+
+            getS3Client().putObject(putObjectRequest, RequestBody.fromBytes(bytes));
+            return key;
+        } catch (S3Exception e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+        }
+        return "";
+    }
+
+    public static byte[] inputStreamToBytes(InputStream inputStream) throws IOException {
```

Comment:
Not sure what happened to my earlier comment, but why are we iterating over bytes when we could use RequestBody.fromInputStream?

Reply:
I will continue to investigate this.